katsdpcontroller/product_controller.py

```diff
@@ -442,10 +458,28 @@ def __init__(self, sched: scheduler.SchedulerBase,
         self.state = ProductState.CONFIGURING   # This sets the sensor
         self.add_sensor(self._capture_block_sensor)
         self.add_sensor(self._state_sensor)
+        # Priority is lower (higher number) than the default queue
+        self.batch_queue = scheduler.LaunchQueue(
+            sdp_controller.batch_role, 'batch', priority=BATCH_PRIORITY)
+        sched.add_queue(self.batch_queue)
+        # generate physical nodes
+        self.physical_graph = self._instantiate_physical_graph(self.logical_graph)
+        # Nodes indexed by logical name
+        self._nodes = {node.logical_node.name: node for node in self.physical_graph}
+        self.telstate_node = self._nodes.get(telstate_name)
+        self.master_controller = sdp_controller.master_controller
+
         logger.info("Created: %r", self)
         logger.info('Logical graph nodes:\n'
                     + '\n'.join(repr(node) for node in self.logical_graph))
 
+    def __del__(self) -> None:
+        if hasattr(self, 'batch_queue'):
+            self.sched.remove_queue(self.batch_queue)
+
+    def __repr__(self) -> str:
+        return "Subarray product {} (State: {})".format(self.subarray_product_id, self.state.name)
```

Comment:
Seems as good a time as any to replace last few remaining `str.format` with f-strings :)

Reply:
True. I can run pyupgrade on it, but I'll make a separate PR for that since it'll generate a lot of simple changes that are a pain to review when mixed in with functional changes.